### DISCUSSION TITLE
Why there is readonly modifier on Table.table?

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#358] [General]
The documentation says: "Note that the table property is marked as readonly, this is needed for TypeScript to check the usage of the table in queries."

```ts
export class UserTable extends BaseTable {
  // Here
  readonly table = "user"

  columns = this.setColumns(...)
}
```

Is this still relevant? Which code triggers errors without `readonly`? I just tried to replace all `readonly table` with `table` on a smaller sized project (5 cross-related tables, but with deeply nested queries), and there were no typecheck errors on Typescript 5.2.2 / orchid-orm 1.18.0.

---

Besides, in the latest Nuxt 3.12 / Typescript 5.5.4 / orchid-orm 1.34.5 project, I'm seeing this when running `nuxt typecheck` (but no errors in the IDE for some reason):

```
server/db/tables/user.table.ts:6:3 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'BaseTableInstance<{ id: () => ColumnWithDefault<UUIDColumn<DefaultSchemaConfig>, () => string> & { data: { primaryKey: string; }; } & { data: { default: RawSQLBase<...>; }; }; ... 52 more ...; timestampsNoTZ<T extends ColumnTypeBase>(this: { ...; }): Timestamps<...>; }>'.
```

I had to add `override readonly table = "user"` to fix that, but now I'm not sure if `readonly` is actually needed?

**Comment by romeerez** [comment]
Ah, I didn't know about `override`, have seen it in bingtsingw's repros and wondered why is that needed.

Removing `readonly` won't cause TS errors, because the `table` property's type will be `string`, and you could reference columns from non-existing tables without objections from TS:

```ts
db.someTable.select('awawa.column')
```

Of course, I wish `readonly` wasn't needed, and will try to redesign table initialization in the future.

**Reply by IlyaSemenov** [reply]
Understood. I see some problems with typings in join queries indeed. Thanks for clarification.
